dotCMS/src/main/java/com/dotmarketing/common/reindex/IndexResource.java

```diff
@@ -0,0 +1,329 @@
+package com.dotmarketing.common.reindex;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.glassfish.jersey.server.JSONP;
+import com.dotcms.content.elasticsearch.business.ClusterStats;
+import com.dotcms.content.elasticsearch.business.ContentletIndexAPI;
+import com.dotcms.content.elasticsearch.business.ESIndexAPI;
+import com.dotcms.content.elasticsearch.business.NodeStats;
+import com.dotcms.content.elasticsearch.util.ESReindexationProcessStatus;
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.rest.InitDataObject;
+import com.dotcms.rest.ResponseEntityView;
+import com.dotcms.rest.WebResource;
+import com.dotcms.rest.annotation.NoCache;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.contentlet.transform.ContentletToMapTransformer;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMap.Builder;
+import com.liferay.portal.language.LanguageException;
+import com.liferay.portal.language.LanguageUtil;
+import io.vavr.control.Try;
+
+@Path("/v1/index")
+public class IndexResource {
+
+    public IndexResource() {
+
+
+    }
+
+
+    private enum IndexAction{
+        ACTIVATE,
+        DEACTIVATE,
+        CLEAR,
+        OPEN,
+        CLOSE;
+        
+        public static IndexAction fromString(String actionStr) {
+            for(IndexAction action: IndexAction.values()) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'action' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)